src/ocean/io/serialize/StructSerializer.d

```diff
@@ -575,12 +575,8 @@ struct StructSerializer ( bool AllowUnions = false )
 
                 static if (isTypedef!(T))
                 {
-                    mixin(`
-                    static if ( is(T B == typedef) )
-                    {
-                        serializer.serialize(data, cast(B)(field), field_name);
-                    }
-                    `);
+                    serializer.serialize(data, cast(StripTypedef!(T))(field),
```

Comment:
To re-iterate previous context - I don't want any non-deprecate symbols in ocean that support typedef as a special concept in D2 mode. That thing is gone and we should focus more on future proof concepts than on legacy support.